### ISSUE TITLE
Sign-in modal is not trapping focus correctly

### ISSUE THREAD

**Issue by artsymartha68** [department-of-veterans-affairs/va.gov-team#108732]
### Origin
This defect was originally found during the VA 508 Office audit for the 1095-B form. Starting with the identity team who owns the sign-in modal, I believe?

### Description
Re: Defect 2: This appears to describe a problem with the sign-in modal itself. For example, click the “Sign in” button in the header of any page. Do shift-tab, and then tab, as described in the audit. If fails anywhere/everywhere, not just on our page.

> Feedback from @SarahKay8 in parent ticket:
> Defect 2 -
> 
> I agree this is a sitewide issue, not specific to this page. The expected behavior is that Shift+Tab cycles back to the modal close button and if it doesnt the modal isn’t trapping focus correctly. Probably should escalate to the Design system team for further discovery
> 
> @artsymartha68 FYI, to be referred to the Design team 

 _Originally posted by @lakper in [#108574](https://github.com/department-of-veterans-affairs/va.gov-team/issues/108574#issuecomment-2840315883)_

**Comment by asg5704** [comment]
@artsymartha68 I just looked at this and have confirmed this is a bug. However, I believe it should be referred to the Design System team for review as it is the `va-modal` component that is not trapping focus so all areas that use `va-modal` will be affected.

I think this requires multiple teams to investigate and find a solution as the `aria-hidden` on the non-modal content is being ignored by the browser (see attached screenshot)

![Image](https://github.com/user-attachments/assets/28351c0f-27a1-48de-acb3-969d1231a629)

Screenshot information
- Page: https://www.va.gov/find-forms/about-form-21-526ez/
- Component used: `va-modal`

**Comment by artsymartha68** [comment]
Thanks for your investigation @asg5704. Adding the DS tags and project info to this ticket.
